### PR TITLE
Feature: toBePartial matcher

### PR DESCRIPTION
**Pull request by YoeriNijs**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
The toBePartial matcher should provide an easy way to combine property assertions. Instead of doing this:

```
const var = { prop1: 'lorem', prop2: 'ipsum', prop3: 'dolor', prop4: 'sit' };
expect(var.prop1).toEqual('lorem');
expect(var.prop2).toEqual('ipsum');
expect(var.prop3).toEqual('dolor');
```

You could do this:

```
const var = { prop1: 'lorem', prop2: 'ipsum', prop3: 'dolor', prop4: 'sit' };
expect(var).toBePartial({ prop1: 'lorem', prop2: 'ipsum', prop3: 'dolor' });
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Did some renaming as well.

### CODE REVIEW

File: projects/spectator/jest/test/matchers/matchers.spec.ts [hunk 1]
```diff
@@ -82,4 +87,21 @@ describe('Matchers', () => {
       expect('#parent-visibility-hidden').toBeHidden();
     });
   });
+
+  describe('toBePartial', () => {
```
Comment (NetanelBasal) [inline projects/spectator/jest/test/matchers/matchers.spec.ts:91]: What about jasmine?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+### Unreleased
```
Comment (NetanelBasal) [inline CHANGELOG.md:5]: Please remove this

File: projects/spectator/src/lib/matchers.ts [hunk 3]
```diff
@@ -339,6 +339,20 @@ export const toBeEmpty = comparator(el => {
   return { pass, message };
 });
 
+/**
+ * Verify if an object has some expected properties.
+ *
+ * const actual = { lorem: 'first', ipsum: 'second' };
+ * expect(actual).toBePartial({ lorem: 'first' });
+ */
+export const toBePartial = comparator((actual, expected) => {
```
Comment (NetanelBasal) [inline projects/spectator/src/lib/matchers.ts:348]: Update the docs, please

File: projects/spectator/jest/test/matchers/matchers.spec.ts (hunk 1 above)
Comment (YoeriNijs) [reply projects/spectator/jest/test/matchers/matchers.spec.ts:91]: Done!

File: projects/spectator/jest/src/lib/matchers-types.ts [hunk 4]
```diff
@@ -34,6 +34,8 @@ declare namespace jest {
 
     toBeEmpty(): boolean;
 
+    toBePartial(partial: any): boolean;
```
Comment (NetanelBasal) [inline projects/spectator/jest/src/lib/matchers-types.ts:37]: Why any? It should a least be an object.

Comment (YoeriNijs) [reply projects/spectator/jest/src/lib/matchers-types.ts]: You are right! Fixed.

**Pull request merged**